v1.4.0

donavanbecker opened the pull request:
## [Version 1.4.0](https://github.com/donavanbecker/homebridge-rainbird/compare/v1.3.0...v1.4.0) (2021-10-28)

### Changes

- Added support to show vavles as contact sensors
    - Allows you to get notified in the Home App when a Zone has started.
- Added Device Logging and Debug Logging for plugin
- Changed from `node-fetch` to `axios`
- Updated dependencies
- Persist configured and duration values
- Fixed issue where some models were not able to stop a zone

donavanbecker committed:
Update codeql-analysis.yml

donavanbecker committed:
Update nodejs-beta.yml

donavanbecker committed:
Update nodejs.yml

mantorok1 committed:
Support Rain Sensor and other improvements

donavanbecker committed:
remove issue validation

mantorok1 committed:
Throw error when invalid CurrentZoneStateResponse

mantorok1 committed:
Fix 'RainBird controller request failed RangeError [ERR_OUT_OF_RANGE]' error for ESP-ME3

donavanbecker committed:
fix spellings

mantorok1 committed:
Merge

donavanbecker committed:
update dependencies

donavanbecker committed:
update dependencies

donavanbecker committed:
change to axios

donavanbecker committed:
fix

donavanbecker committed:
remove un necessary stuff

donavanbecker committed:
update dependencies

donavanbecker committed:
update dependencies

donavanbecker committed:
minor logging and naming changes

donavanbecker committed:
done

mantorok1 committed:
Merge branch 'beta' of github.com:donavanbecker/homebridge-rainbird into beta

mantorok1 committed:
Fix axios request

donavanbecker committed:
add contact sensors

donavanbecker committed:
make corrections for contact sensor to recognize config

donavanbecker committed:
Update platform.ts

mantorok1 commented on config.schema.json:
```diff
@@ -61,15 +67,31 @@
-            "description": "Turn plugin's debug mode on."
+            "title": "Logging Settings",
+            "type": "string",
+            "description": "Enble 'Device' or 'Debug' logging.",
```
Typo: Enble should be Enable

mantorok1 reviewed (changes requested):
Looks good overall, just a minor typo

donavanbecker committed:
corrections to contact sensors

donavanbecker committed:
potential fix for contact sensors

donavanbecker committed:
potential fix for contact sensors

donavanbecker committed:
potential fix for contact sensors

donavanbecker committed:
Correction

donavanbecker committed:
Update ContactSensor.ts

donavanbecker committed:
more corrections

donavanbecker committed:
Update config.schema.json

mantorok1 committed:
Fix contact sensors

mantorok1 committed:
Fix stop zone not working for some models

mantorok1 committed:
Add request queue

mantorok1 committed:
Persist configured and duration values

donavanbecker committed:
v1.4.0

## [Version 1.4.0](https://github.com/donavanbecker/homebridge-rainbird/compare/v1.3.0...v1.4.0) (2021-10-28)

### Changes

- Added support to show vavles as contact sensors
    - Allows you to get notified in the Home App when a Zone has started.
- Added Device Logging and Debug Logging for plugin
- Changed from `node-fetch` to `axios`
- Updated dependencies
- Persist configured and duration values
- Fixed issue where some models were not able to stop a zone
